core/templates/pages/teach-page/teach-page.component.ts

```diff
@@ -16,64 +16,62 @@
  * @fileoverview Component for the teach page.
  */
 
-require('base-components/base-content.directive.ts');
-require(
-  'components/common-layout-directives/common-elements/' +
-  'background-banner.component.ts');
+import { Component, OnInit } from '@angular/core';
+import { downgradeComponent } from '@angular/upgrade/static';
 
-require('domain/utilities/url-interpolation.service.ts');
-require('services/site-analytics.service.ts');
+import { UrlInterpolationService } from
+  'domain/utilities/url-interpolation.service';
+import { SiteAnalyticsService } from 'services/site-analytics.service';
+import { WindowRef } from 'services/contextual/window-ref.service';
 
-angular.module('oppia').component('teachPage', {
-  template: require('./teach-page.component.html'),
-  controller: [
-    '$timeout', 'SiteAnalyticsService',
-    'UrlInterpolationService', 'WindowRef',
-    function(
-        $timeout, SiteAnalyticsService,
-        UrlInterpolationService, WindowRef) {
-      const ctrl = this;
-      // Define constant for each tab on the page.
-      ctrl.TAB_ID_TEACH = 'teach';
-      ctrl.TAB_ID_PARTICIPATION = 'participation';
-      ctrl.TEACH_FORM_URL = 'https://goo.gl/forms/0p3Axuw5tLjTfiri1';
-
-      const ALLOWED_TABS = [
-        ctrl.TAB_ID_TEACH, ctrl.TAB_ID_PARTICIPATION];
-
-      ctrl.activeTabName = ctrl.TAB_ID_TEACH;
-
-      ctrl.onTabClick = function(tabName) {
-        // ---- Update hash ----
-        WindowRef.nativeWindow.location.hash = '#' + tabName;
-        ctrl.activeTabName = tabName;
-      };
-
-      ctrl.getStaticImageUrl = function(imagePath) {
-        return UrlInterpolationService.getStaticImageUrl(imagePath);
-      };
+@Component({
+  selector: 'teach-page',
+  templateUrl: './teach-page.component.html',
+  styleUrls: []
+})
+export class TeachPageComponent implements OnInit {
+  TAB_ID_TEACH: string = 'teach';
+  TAB_ID_PARTICIPATION: string = 'participation';
+  TEACH_FORM_URL: string = 'https://goo.gl/forms/0p3Axuw5tLjTfiri1';
+  ALLOWED_TABS: string[] = [this.TAB_ID_TEACH, this.TAB_ID_PARTICIPATION];
+  activeTabName: string = this.TAB_ID_TEACH;
+  constructor(
+    private siteAnalyticsService: SiteAnalyticsService,
+    private urlInterpolationService: UrlInterpolationService,
+    private windowRef: WindowRef
+  ) {}
+  ngOnInit(): void {
```

Comment:
Add a line break ahead of each function.

Reply:
Done